### DISCUSSION TITLE
Display Currency (front end)

### DISCUSSION THREAD

**Discussion by daviroo** [GaloyMoney/galoy-mobile#707] [General]
This is a place for people to discuss the display currency feature which we will start implementing in the mobile app.  The backend work for this hasn't been done yet but we have decided to start implementing the front end first.  First thing I would like to get clarity on is what is part of the front end - in my head I'm thinking the currency conversion would happen on the backend.  So the mobile app won't be querying APIs for exchange rates for different currency pairs.  Instead this data will be returned from the API, perhaps the current exchange rate as part of the price subscription and also some sort of historical conversion value in each transaction returned from the API.

**Comment by vindard** [comment]
Relevant backend discussion to support this: https://github.com/GaloyMoney/galoy/discussions/1280#discussioncomment-2941523

**Reply by daviroo** [reply]
> Relevant backend discussion to support this: [GaloyMoney/galoy#1280 (comment)](https://github.com/GaloyMoney/galoy/discussions/1280#discussioncomment-2941523)

Great - I didn't realise this one already existed.  Seems you had similar thoughts to me on having a historical rate associated with transactions.  What about the idea of including the current rate in the price subscription?  This would be so we could display accurate current balances.  We would need the rate for display currency/BTC and display currency/USD to enable the unified balance on the front end.

**Reply by vindard** [reply]
Yea I think that makes sense. We potentially could also return the `balance` in the `me` query in the currently set display currency.

**Comment by nicolasburtey** [comment]
the first 2 PR I would do:
- make sure every place in the mobile app where there is currently a dollar become templated, so that you can pass a currency on the component, and it renders with the right currency
- similar to the language, make it possible to save the `current display currency` in the backend (probably at the Account level, not at the user level). there would need to a be a query to store and retreive this value from the graphql.

when this second PR is been done, now we can do another PR in the backend to save the current `display currency` on each `transaction` and return this as part of the transaction.

**Comment by ntheile** [comment]
Here is some interesting libraries that I have stumbled upon for exchange rates:

- **Open Exchange Rates**: used by big companies like Kickstarter and shopify, has historical pricing and supports hundreds of currencies https://openexchangerates.org/
- **Money.js**: SDK for openexchange rates http://openexchangerates.github.io/money.js/ 
- **Accounting.js**: lib for currency formatting http://openexchangerates.github.io/accounting.js/
